utils/micro-controllers.stanza

```diff
@@ -130,68 +131,20 @@ public defn MicroController (properties:Tuple<KeyValue>, exist:Tuple<String>, so
 ;============================================================
 ;======================= MCU to JITX ========================
 ;============================================================
-
-defmethod to-jitx (mcu: MicroController) -> Instantiable :
-  pcb-component my-microcontroller :
-    manufacturer = manufacturer(mcu)
-    mpn = mpn(mcu)
-
-    to-jitx-pin-properties(pin-properties(mcu))
-    val bundle-table = to-jitx-bundles(bundles(mcu))
-    to-jitx-supports(supports(mcu), bundle-table)
-
-    assign-landpattern $
-      switch(mfg-package(mcu)) :
-        "LQFP100" : LQFP100
-        "LQFP128" : LQFP128
-        "LQFP144" : LQFP144
-        "LQFP176" : LQFP176
-        "LQFP208" : LQFP208
-        "LQFP48" : LQFP48
-        "UFQFPN28" : UFQFPN28
-        "UFQFPN32" : UFQFPN32
-        "UFQFPN48" : UFQFPN48
-        "LFBGA100" : LFBGA100
-        "LFBGA144" : LFBGA144
-        "LFBGA354" : LFBGA354
-        "TFBGA100" : TFBGA100
-        "TFBGA216" : TFBGA216
-        "TFBGA225" : TFBGA225
-        "TFBGA64" : TFBGA64
-        "UFBGA100" : UFBGA100
-        "UFBGA121" : UFBGA121
-        "UFBGA129" : UFBGA129
-        "UFBGA132" : UFBGA132
-        "UFBGA144" : UFBGA144
-        "UFBGA169" : UFBGA169
-        "UFBGA64" : UFBGA64
-        "UFBGA73" : UFBGA73
-        "WLCSP100" : WLCSP100
-        "WLCSP104" : WLCSP104
-        "WLCSP115" : WLCSP115
-        "WLCSP132" : WLCSP132
-        "WLCSP143" : WLCSP143
-        "WLCSP144" : WLCSP144
-        "WLCSP156" : WLCSP156
-        "WLCSP168" : WLCSP168
-        "WLCSP18" : WLCSP18
-        "WLCSP180" : WLCSP180
-        "WLCSP20" : WLCSP20
-        "WLCSP25" : WLCSP25
-        "WLCSP36" : WLCSP36
-        "WLCSP49" : WLCSP49
-        "WLCSP52" : WLCSP52
-        "WLCSP63" : WLCSP63
-        "WLCSP64" : WLCSP64
-        "WLCSP66" : WLCSP66
-        "WLCSP72" : WLCSP72
-        "WLCSP81" : WLCSP81
-        "WLCSP90" : WLCSP90
-        else :
-          fatal("MCU package %_ is not supported." % [mfg-package(mcu)])
-    make-box-symbol()
-  my-microcontroller
-
+pcb-component mcu-component (manu:String, pn:String, pp:STMPinProperties, supports:Tuple<STMSupports>, bundles:Tuple<STMBundle>, pkg:String) :
```

Comment:
The component generation can now be cached

Reply:
Quite genius!

Reply:
Perhaps instead of `manu` and `pn` which aren't immediately clear:
`mcu-manufacturer` and `mcu-mpn`

Reply:
good point, fixed